versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -234,17 +224,18 @@ public Hash merge(
                 writeRefLogEntry(
                     ctx,
                     pointer,
-                    toBranch.getName(),
+                    mergeParams.getToBranch().getName(),
                     RefLogEntry.RefType.Branch,
-                    toHead,
+                    newHead,
                     RefLogEntry.Operation.MERGE,
                     timeInMicros,
-                    Collections.singletonList(from));
+                    Collections.singletonList(mergeParams.getMergeFromHash()));
 
             // Return hash of last commit (toHead) added to 'targetBranch' (via the casOpLoop)
-            return updateGlobalStatePointer(toBranch, pointer, toHead, newGlobalHead, newRefLog);
+            return updateGlobalStatePointer(
+                mergeParams.getToBranch(), pointer, newHead, newGlobalHead, newRefLog);
           },
-          () -> mergeConflictMessage("Retry-failure", from, toBranch, expectedHead));
+          () -> mergeConflictMessage("Retry-failure", mergeParams));
```

Comment:
tangential: should we perhaps report `currentHead` too?

Reply:
not here (this is "just" a retry) - but in general, yes: #3853